alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,417 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"strings"
+	"time"
+
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/polardb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudPolarDBCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudPolarDBClusterCreate,
+		Read:   resourceAlicloudPolarDBClusterRead,
+		Update: resourceAlicloudPolarDBClusterUpdate,
+		Delete: resourceAlicloudPolarDBClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_type": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_version": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_node_class": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cluster_charge_type": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{string(Postpaid), string(Prepaid)}, false),
+				Optional:     true,
```

Comment:
If the `cluster_charge_type` does not support updating, please set it to `ForceNew`. Still applies for other parameters.

Reply:
done